New RSTUF Guide (#341)
repository-service-tuf/repository-service-tuf: docs/source/guide/deployment/guide/kubernetes.rst

##########
Kubernetes
##########

> You can deploy multiple RSTUF API instances/replicas in a distributed
> environment to support multiple requests.

Reviewer: ```suggestion
Multiple RSTUF API instances/replicas can be deployed in a distributed
environment to support multiple requests.
```

This example uses Kubernetes

.. note::

>   This deployment guide doesn't show the Kubernetes cluster configuration.
>   It requires a Kubernetes cluster running with ``kubectl`` configured.

Reviewer: ```suggestion
  This deployment guide does not show the Kubernetes cluster configuration.
  It requires a Kubernetes cluster running with ``kubectl`` configured.
```

.. note::

>   For this deployment we will use Redis as the
>   :ref:`message queue/broker <guide/deployment/planning/deployment:Message Queue/Broker: Redis or RabbitMQ server>`.

Reviewer: ```suggestion
  For this deployment Redis is used as the
  :ref:`message queue/broker <guide/deployment/planning/deployment:Message Queue/Broker: Redis or RabbitMQ server>`.
```

.. Warning::

>   This deployment doesn't have :ref:`guide/deployment/planning/deployment:Authentication/Authorization`
>   for the API.

Reviewer: ```suggestion
  This deployment does not have :ref:`guide/deployment/planning/deployment:Authentication/Authorization`
  for the API.
```

  This API is fully accessible for unauthorized users. Consider using an
  API :ref:`guide/deployment/planning/deployment:Authentication/Authorization`
  service.

.. note::
  See the complete :ref:`guide/deployment/planning/deployment:Deployment Configuration`
  for in-depth details.

Requirements
############

Software and tools
==================

  * Kubernetes Cluster
  * kubectl
  * Python >= 3.10
  * pip

Online Key
##########

This deployment requires the :ref:`guide/general/keys:Online Key`.

> See the chapter :ref:`guide/general/keys:Singing Keys`

Reviewer: signing keys ??

> .. centered:: You can skip it if you already have the online key.

Reviewer: ```suggestion
.. centered:: Skip if the online key has been secured.
```

RSTUF Command Line Interface (CLI) provides a feature for
:ref:`guide/repository-service-tuf-cli/index:Key Generation (``generate\`\`)`

.. include:: ../../repository-service-tuf-cli/index.rst
    :start-after: Key Generation (``generate``)

Secrets
#######

.. caution::

>     Do not use this credentials, it is just an example.

Reviewer: ```suggestion
    Do not use this credential, it is just an example.
```

onlinekey secret
================

Generating the secrets for RSTUF Online Key

> 1. Get the key content and transforms encode it to base 64

Reviewer: ```suggestion
1. Get the key content and encode it to base 64
```
